tests/check_framework/test_templates.py

```diff
@@ -78,30 +93,36 @@ def setUpClass(cls):
             string_if_invalid2, type(string_if_invalid2).__name__
         )
 
+    @staticmethod
+    def _check_all_engines():
```

Comment:
`@staticmethod` should [generally be avoided](https://www.seporaitis.net/posts/2020/05/05/python-staticmethod-usage/):

```
    def _check_all_engines(self):
```

Reply:
Noted, I was not aware of that.